scio-core/src/main/scala/com/spotify/scio/io/CompressionType.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.io
+
+import java.io.{InputStream, OutputStream}
+
+import com.google.cloud.dataflow.sdk.io.TextIO.{CompressionType => GCompressionType}
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+
+/**
+ * Wrapper for [[com.google.cloud.dataflow.sdk.io.TextIO.CompressionType]] with
+ * helper methods for reading and writing.
+ */
+sealed trait CompressionType {
```

Comment:
Prefer not to add any more public interface right now. So remove this and move the wrap logic to `TapTest`?